[ADD] maintain: add google oauth doc (#2274)
odoo/documentation: content/administration/maintain/google_oauth.rst

---------------------------------------

Choosing :guilabel:`Production` as the :guilabel:`Publishing Status` (instead of Testing) will
display the following warning message:

.. image:: google_oauth/published-status.png
   :class: img-fluid
   :align: center
   :alt: OAuth is Limited to 100 Sensitive Scope Logins.

To correct this warning, navigate to the
`Google API Platform <https://console.cloud.google.com/apis/credentials/consent>`__. If the
:guilabel:`Publishing status` is :guilabel:`In Production`, click :guilabel:`Back to Testing`
to correct the issue.

No Test Users Added
-------------------

If no test users are added to the OAuth consent screen, then a 403 access denied error will
populate.

.. image:: google_oauth/403-error.png
   :class: img-fluid
   :align: center
   :alt: 403 Access Denied Error.

To correct this error, return to the :guilabel:`OAuth consent screen` under
:guilabel:`APIs & Services` and add test user(s) to the app. Add the email that you want to
configure in Odoo.

Gmail Module not updated
------------------------

If the :guilabel:`Gmail Module` in Odoo has not be updated to the latest version, then a
:guilabel:`Forbidden` error message populates.

> .. admonition:: Forbidden

Reviewer: ![forbidden-error](https://user-images.githubusercontent.com/91896312/177778480-a1a0551b-a978-49a5-84dc-8ffb37989a47.png)

Here is the updated screenshot.

========================================
Connect Gmail to Odoo using Google OAuth
========================================

Odoo is compatible with Google's OAuth for Gmail. In order to send secure emails from a customd
domain, configure a few settings on Google's *Workspace* platform, as well as on the back end of
the Odoo database. This is necessary to use a Google external email server with both a personal

> (yourname\@gmail.com) and a custom domain (name\@yourcompany.com / Google Workspace Account).

Reviewer: Thanks for making these changes- It is much more understandable now. I like how you added in "or".

.. tip::
   For more information, visit
   `Google's documentation <https://support.google.com/cloud/answer/6158849>`__ on setting up
   OAuth.

Setup in Google
===============

Create a new project
--------------------

To get started, go to the `Google API Platform
<https://console.cloud.google.com/apis/credentials/consent>`__. Log in with your *Google Workspace*
account if you have one, otherwise log in with your personal Gmail account. (Choose the email

> address you want to configure in Odoo.)

Reviewer: Looks Good!

After that, click on :guilabel:`Create Project`, located on the far right of the OAuth consent
screen. If a project has already been created in this account, then the :guilabel:`New Project`
option will be located on the top left under the :guilabel:`Select a project` drop-down menu.

> On the :guilabel:`New Project` screen, rename the :guilabel:`Project` to *Odoo* and browse for the

Reviewer: Agreed-- looks better now.

:guilabel:`Location`. Set the :guilabel:`Location` as the *Google Workspace* organization. If you
are using a personal Gmail account, then leave the :guilabel:`Location` as *No Organization*.

.. image:: google_oauth/new-project.png
   :class: img-fluid
   :align: center
   :alt: Project Name and Location for Google Oauth.

Next, click on :guilabel:`Create`, and then the :guilabel:`User Type` screen appears. Select the
:guilabel:`External` option, and then click on :guilabel:`Create` again. Then, the page redirects

> to the :guilabel:`Edit app registration` page.

Reviewer: Sounds better!

- Next, add the email being configured under the :guilabel:`Test users` step by clicking on
  :guilabel:`Add Users` and then the :guilabel:`Save and Continue` button. A summary of the
  :guilabel:`App registration` appears.
- Finally, scroll to the bottom and click on :guilabel:`Back to Dashboard`.

Create Credentials
------------------

Now that you have set up the project, you need to create credentials, which includes the
:guilabel:`Client ID` and :guilabel:`Client Secret`. First, click on :guilabel:`Credentials` in the
left sidebar menu.

Then, click on :guilabel:`Create Credentials` in the top menu and select
:guilabel:`OAuth client ID` from the dropdown menu.

- Under :guilabel:`Application Type`, select :guilabel:`Web Application` from the dropdown menu.
- In the :guilabel:`Client ID` field, enter `Odoo`.
- Under the :guilabel:`Authorized redirect URIs` label, click the button :guilabel:`ADD URI`, and
  then input `https://yourdbname.odoo.com/google_gmail/confirm` in the :guilabel:`URIs 1` field. Be
  sure to replace the *yourdbname* part of the URL with your actual Odoo database name.
- Next, click on :guilabel:`Create` to generate an OAuth :guilabel:`Client ID` and

>   :guilabel:`Client Secret`. Finally, copy each generated value to a Notepad and navigate to the

Reviewer 2: Notepad as a proper noun assumes the user is using the Windows application.
Suggestion below, but not required.

```suggestion
  :guilabel:`Client Secret`. Finally, copy each generated value for later use when configuring in Odoo, and then navigate to the
```

  Odoo database.

.. image:: google_oauth/client-credentials.png
   :class: img-fluid
   :align: center
   :alt: Client ID and Client Secret for Google OAuth.

Setup in Odoo
=============

Enter Google Credentials
------------------------

First, open Odoo and navigate to the :guilabel:`Apps` module. Then, remove the
:guilabel:`Apps` filter from the search bar and type in `Google`. Install the module called
:guilabel:`Google Gmail`.

> Next, navigate to the :menuselection:`Settings module --> General Settings`, and under the

Reviewer 2: **Settings module** is not the name of the UI element 😉.

Suggestion below, not required.

```suggestion
Next, navigate to :menuselection:`Settings --> General Settings`, and under the
```

Next, navigate to the :menuselection:`Settings module --> General Settings`, and under the
:guilabel:`Discuss` section, ensure that the checkbox for :guilabel:`Custom Email Servers` is

> checked. This populates a new option for :guilabel:`Gmail Credentials`. Make sure to check the box
> next to it. Then, copy and paste the :guilabel:`Client ID` and :guilabel:`Client Secret` into the

Reviewer: Lets remove the line that reads: "Make sure to check the box next to it."

I don't know if the formatting is correct below. 

```suggestion
checked. This populates a new option for :guilabel:`Gmail Credentials`. 

Then, copy and paste the :guilabel:`Client ID` and :guilabel:`Client Secret` into the
```

After that, on the :guilabel:`General Settings` page, under :guilabel:`Custom Email Servers`, click
:guilabel:`Outgoing Email Servers` to configure the external Gmail account.

.. image:: google_oauth/outgoing-servers.png
   :class: img-fluid
   :align: center
   :alt: Configure Outgoing Email Servers in Odoo.

Then, :guilabel:`Create` a new email server and select the option for :guilabel:`Gmail`. Next, fill
in the :guilabel:`Description` (can be anything) and the email :guilabel:`Username` and click on
:guilabel:`Connect your Gmail account`.

A new window from :guilabel:`Google` appears to complete the authorization process. Select the
appropriate email address that is being configured in Odoo.

If the email address is a personal account, an extra step pops up, click :guilabel:`Continue` to
allow the verification and connect the Gmail account to Odoo.

Then, allow Odoo to access the Google account by clicking on :guilabel:`Continue` or
:guilabel:`Allow`. After that, the page navigates back to the newly configured outgoing email
server in Odoo. The configuration automatically loads the token in Odoo, and a tag stating
:guilabel:`Gmail Token Valid` appears in green.

> .. image:: google_oauth/green-token.png

Reviewer 3: This image is too wide at 1430 px. Can you retake this screenshot @tiku-odoo ? You will probably have to shorten the width of the browser window a bit more to keep it under the tablet breakpoint width (768 px). And remember to compress your PNG image file after you take the screenshot!

---------------------------------------

Choosing :guilabel:`Production` as the :guilabel:`Publishing Status` (instead of Testing) will
display the following warning message:

.. image:: google_oauth/published-status.png
   :class: img-fluid
   :align: center
   :alt: OAuth is Limited to 100 Sensitive Scope Logins.

To correct this warning, navigate to the
`Google API Platform <https://console.cloud.google.com/apis/credentials/consent>`__. If the
:guilabel:`Publishing status` is :guilabel:`In Production`, click :guilabel:`Back to Testing`
to correct the issue.

No Test Users Added
-------------------

If no test users are added to the OAuth consent screen, then a 403 access denied error will
populate.

.. image:: google_oauth/403-error.png
   :class: img-fluid
   :align: center
   :alt: 403 Access Denied Error.

To correct this error, return to the :guilabel:`OAuth consent screen` under
:guilabel:`APIs & Services` and add test user(s) to the app. Add the email that you want to
configure in Odoo.

Gmail Module not updated
------------------------

If the :guilabel:`Gmail Module` in Odoo has not be updated to the latest version, then a
:guilabel:`Forbidden` error message populates.

> .. image:: google_oauth/forbidden-error.png

Reviewer 3: This screenshot is too wide at 988px. Can you retake this screenshot @tiku-odoo ? And remember to compress the PNG image file after you take the screenshot!

========================================
Setup Google Gmail using OAuth with Odoo

> ========================================

Reviewer 4: Ah maybe this, though!
No need to add "with Odoo". Else, all titles would have it :D

Odoo is compatible with Google's OAuth for Gmail. With only a few steps to set up on Google's
platform, and a few steps to configure in the Odoo database, sending emails securely from a custom
domain (on *Google Workspace*) to customers is easy! Here's `Google's documentation

> <https://support.google.com/cloud/answer/6158849>`__ on setting up OAuth.

Reviewer 2: Second thought, might be better as a `:seemore:` tag actually...

> <https://support.google.com/cloud/answer/6158849>`__ on setting up OAuth.

Reviewer 4: Agreed :) 
I don't know if seemore works, but I'm sure `seealso` will do ^^

Valid* will appear in green.

.. image:: google_oauth/gmail-token-valid.png
   :align: center
   :class: img-fluid

Save the settings and *Test the Connection*.

The Google Gmail account is now functioning in Odoo using OAuth to authenticate. The database is
now sending safe, secure emails through Google using Oauth authentication.

Google OAuth FAQ
================

**Production** VS **Testing** *Publishing Status*
-------------------------------------------------

Choosing *Production* as the *Publishing Status* (instead of *Testing*) will display the following
warning message:

.. warning::
   OAuth is limited to 100 sensitive scope logins until the OAuth consent screen is verified. This
   may require a verification process that can take several days.

To correct this warning, navigate to the `Google APIs platform
<https://console.cloud.google.com/apis/credentials/consent>`__. If in *Production*, then click
*Back to Testing* to correct the issue.

.. image:: google_oauth/back-to-testing.png
   :align: center
   :class: img-fluid

**Desktop App** VS **Web Application** *Application Type*
---------------------------------------------------------

> Should *Desktop App* under the *Application Type* be selected when creating the *Credentials (OAuth
> client ID)*, the following error message will display:
> 
> .. image:: google_oauth/error-400.png
>    :align: center
>    :class: img-fluid

Reviewer 2: Add `:guilabel:`'s here

Also, @tiku-odoo we should avoid using screenshots as primary means of communicating information. Screenshots only serve as supplementary context to support the text.

In this section here, someone who has poor internet connection or is visually impaired can not see what the warning is because there is no written context and alt text is missing.

```suggestion
Should :guilabel:`Desktop App` under the :guilabel:`Application Type` be selected when creating
the :guilabel:`Credentials (OAuth client ID)`, the following error message will display:

.. image:: google_oauth/error-400.png
   :align: center
   :class: img-fluid
   :alt: OAuth error due to incorrect application type
   
```

To correct this error, delete the credentials already created and create new credentials using the

> *'Web Application' Application Type*.

Reviewer 2: Not sure how to format this because it's not clear what these are @tiku-odoo 
Are these dashboards, sections, labels?

Likely need `:guilabel:`'s here but this is why contextual copy is important. We're assuming the reader knows what these mean...

.. image:: google_oauth/web-application.png
   :align: center

>    :class: img-fluid

Reviewer 2: Line 200, missing `alt` text.

> Under *Authorized redirect URIs*,  ADD URI: https://yourdbname.odoo.com/google_gmail/confirm
> 
> .. note::
>    Replace “yourdbname” in the URL with the Odoo database name.

Reviewer 2: Fixed formatting, added contextual language, and removed **note**.
```suggestion
Under the :guilabel:`Authorized redirect URIs` section, find the :guilabel:`ADD URI` field and
enter `https://yourdbname.odoo.com/google_gmail/confirm`, being sure to replace `yourdbname` in the
URL with your Odoo database name.
```

> .. note::
>    `Google’s documentation <https://support.google.com/cloud/answer/6158849#zippy=%2Cnative-applica
>    tions%2Cweb-applications>`__ on Web applications vs. Native applications.

Reviewer 2: I think this would better be formatted as a **See Also** tag.
```suggestion
.. seealso::
   `Google’s documentation <https://support.google.com/cloud/answer/6158849#zippy=%2Cnative-applica
   tions%2Cweb-applications>`__ on Web applications vs. Native applications.
```

========================================
Connect Gmail to Odoo using Google OAuth
========================================

Odoo is compatible with Google's OAuth for Gmail. In order to send secure emails from a custom
domain, configure a few settings on Google's *Workspace* platform, as well as on the back end of
the Odoo database. This is necessary to use a Google external email server with both a personal

> (youname@gmail.com) and custom domain (name@yourcompany.com / Google Workspace Account).

Reviewer 3: Line 8: you -> your; change `(youname@gmail.com)` to `(yourname@gmail.com)`

Reviewer 3: Also, the fake emails in Line 8, when this RST file is rendered to the HTML page, are highlighted and have external link buttons so you can click on them to send an email to the fake emails, not sure if that can be fixed, but it would be nice if they were just plain text.

Author: @mivu-odoo Great point! I was able to prevent formatting the emails by escaping the @ character.  
For example: `example\@gmail.com`

[More info on escaping text](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#escaping-mechanism)

.. tip::
   For more information, visit
   `Google's documentation <https://support.google.com/cloud/answer/6158849>`__ on setting up
   OAuth.

Setup in Google
===============

Create a new project
--------------------

> To get started, go to `Google APIs platform

Reviewer 3: * add `the` before `Google APIs platform`  
* change `APIs` to singular `API` and capitalize the first letter in `platform`

> <https://console.cloud.google.com/apis/credentials/consent>`__ Log in with your *Google Workspace*

Reviewer 3: Add a period after "<https://console.cloud.google.com/apis/credentials/consent>`__ "

> account if you have one, otherwise log in with your personal Gmail account. Choose the email

Reviewer 3: Line 23-24: add parentheses around `Choose the email address you want to configure in Odoo.`

address you want to configure in Odoo.

After that, click on :guilabel:`Create Project`, located on the far right of the OAuth consent

> screen. If a project has already been created in this account then the new project option will be

Reviewer 3: * add a comma after `If a project has already been created in this account`
* add a `:guilabel:` to `new project` and capitalize first letters in `new project`

located on the top left under the :guilabel:`Select a project` drop-down menu.

> On the new project screen, rename the :guilabel:`Project` to *Odoo* and browse for the

Reviewer 3: Line 30: add a `:guilabel:` to `new project` and capitalize first letters in `new project`

:guilabel:`Location`. Set the :guilabel:`Location` as the *Google Workspace* organization. If you
are using a personal Gmail account, then leave the :guilabel:`Location` as *No Organization*.

.. image:: google_oauth/new-project.png
   :class: img-fluid
   :align: center
   :alt: Project Name and Location for Google Oauth.

Next, click on :guilabel:`Create`, and then the :guilabel:`User Type` screen appears. Select the

> :guilabel:`External` option, and then click on :guilabel:`Create`.

Reviewer 3: * add `again` after `:guilabel:'Create'`
* add a sentence at the end (after `then click on :guilabel:'Create'.`), `Then, the page redirects to the :guilabel:'Edit app registration' page.`

- Next, under :menuselection:`App Domain --> Authorized domains`, click on :guilabel:`Add Domain` and
  enter *odoo.com*.
- After that, under the :guilabel:`Developer contact information` section, enter the organization's
  email address. Google uses this email address to notify the organization about any changes to
  your project.
- Next, click on the :guilabel:`Save and Continue` button.
- Then, skip the :guilabel:`Scopes` page by scrolling to the bottom and clicking on
  :guilabel:`Save and Continue`.
- Next, add the email being configured under the :guilabel:`Test users` step by clicking on
  :guilabel:`Add Users` and then the :guilabel:`Save and Continue` button. A summary of the
  :guilabel:`App registration` appears.
- Finally, scroll to the bottom and click on :guilabel:`Back to Dashboard`.

Create Credentials
------------------

Now that you have set up the project, you need to create credentials, which includes the
:guilabel:`Client ID` and :guilabel:`Client Secret`. First, click on :guilabel:`Credentials` in the
left sidebar menu.

Then, click on :guilabel:`Create Credentials` in the top menu and select
:guilabel:`OAuth client ID` from the dropdown menu.

> .. image:: google_oauth/create-credentials.png

Reviewer 3: This image can be deleted.

Reviewer 3: Looks like this one might have been accidentally missed! Please delete Lines 72-76 and delete the image file from the file tree by using `git rm image-file-path`

   :class: img-fluid
   :align: center
   :alt: Create Credentials on Google OAuth.

- Under :guilabel:`Application Type`, select :guilabel:`Web Application` from the dropdown menu.
- In the :guilabel:`Client ID` field, enter *Odoo*.
- Under the :guilabel:`Authorized redirect URIs` label, click the button :guilabel:`ADD URI`, and
  then input *https://yourdbname.odoo.com/google_gmail/confirm* in the :guilabel:`URIs 1` field. Be
  sure to replace the *yourdbname* part of the URL with your actual Odoo database name.

> .. image:: google_oauth/add-uri.png

Reviewer 3: This image can be deleted.

   :class: img-fluid
   :align: center
   :alt: Add URI in Creating Credentials for Google OAuth.

> - Next, click on :guilabel:`Create` and Google will generate an OAuth :guilabel:`Client ID` and

Reviewer 3: Line 93: replace `and Google will` with `to`

  :guilabel:`Client Secret`. Finally, copy each generated value to a Notepad and navigate to the
  Odoo database.

.. image:: google_oauth/client-credentials.png
   :class: img-fluid
   :align: center
   :alt: Client ID and Client Secret for Google OAuth.

Setup in Odoo
=============

Enter Google Credentials
------------------------

First, open Odoo and navigate to the :guilabel:`Apps` module. Then, remove the
:guilabel:`Apps filter` and type in *Google*. Install the module called :guilabel:`Google Gmail`.

Next, navigate to the :menuselection:`Settings module --> General Settings`, and under the
:guilabel:`Discuss` section, ensure that the checkbox for :guilabel:`Custom Email Servers` is

> checked. This populates a new option for :guilabel:`Gmail Credentials`. Here, copy and paste the

Reviewer 3: * after `:guilabel:'Gmail Credentials'.`, add a sentence `Make sure to check the box next to it.` 
* replace `Here` with `Then`

:guilabel:`Client ID` and :guilabel:`Client Secret` into the respective fields and :guilabel:`Save`
the settings.

> .. image:: google_oauth/google-credentials.png

Reviewer 3: This image can be deleted.

   :align: center
   :alt: Gmail Credentials in Odoo for Google OAuth.

Configure outgoing email server
-------------------------------

After that, on the :guilabel:`General Settings` page, under :guilabel:`Custom Email Servers`, click
:guilabel:`Outgoing Email Servers` to configure the external Gmail account.

.. image:: google_oauth/outgoing-servers.png
   :class: img-fluid
   :align: center
   :alt: Configure Outgoing Email Servers in Odoo.

> Then, :guilabel:`create` a new email server and select the option for :guilabel:`Gmail`. Next, fill

Reviewer 3: Line 132: capitalize the first letter in `create`

in the :guilabel:`Description` (can be anything) and the email :guilabel:`Username` and click on
:guilabel:`Connect your Gmail account`.

A new window from :guilabel:`Google` appears to complete the authorization process. Select the
appropriate email address that is being configured in Odoo.

If the email address is a personal account, an extra step pops up, click :guilabel:`Continue` to
allow the verification and connect the Gmail account to Odoo.

> .. image:: google_oauth/google-unverified.png

Reviewer 3: This image can be deleted.

   :class: img-fluid
   :align: center
   :alt: Google Hasn't Verified This App.

> Finally, allow Odoo to access the Google account by clicking on :guilabel:`Continue` or

Reviewer 3: * delete the line breaks until Line 147 is in the same paragraph as Line 140
* replace `Finally` with `Then`

:guilabel:`Allow`.

After that, the page navigates back to the newly configured outgoing email server in Odoo. The

> configuration automatically loads the token in Odoo, and a graphic stating

Reviewer 3: Line 151: replace `graphic` with `tag`

:guilabel:`Gmail Token Valid` appears in green.

.. image:: google_oauth/green-token.png
   :class: img-fluid
   :align: center
   :alt: Configure Outgoing Email Servers in Odoo.

Finally, save the settings and :guilabel:`Test the Connection`.

> 

Reviewer 3: Line 160: delete this line break until Line 159 and 161 are in the same paragraph

The Odoo database can now send safe, secure emails through Google using OAuth authentication.

Google OAuth FAQ
================

Production VS Testing Publishing Status
---------------------------------------

Choosing :guilabel:`Production` as the :guilabel:`Publishing Status` (instead of Testing) will
display the following warning message:

.. image:: google_oauth/published-status.png
   :class: img-fluid
   :align: center
   :alt: OAuth is Limited to 100 Sensitive Scope Logins.

To correct this warning, navigate to the 

> `Google APIs platform <https://console.cloud.google.com/apis/credentials/consent>`__. If the app is

Reviewer 3: * replace `Google APIs platform` with `Google API Platform`
* replace `app` with `guilabel:'Publishing status'`

> in :guilabel:`Production`, then click :guilabel:`Back to Testing` to correct the issue.

Reviewer 3: * replace `in :guilabel:'Production'` with `:guilabel:'In Production'`
* delete the word `then`

> .. image:: google_oauth/back-to-testing.png

Reviewer 3: This image can be deleted.

   :class: img-fluid
   :align: center
   :alt: Back to Testing in Google OAuth.

No Test Users Added
-------------------

If no test users are added to the OAuth consent screen, then a 403 access denied error will
populate.

.. image:: google_oauth/403-error.png
   :class: img-fluid
   :align: center
   :alt: 403 Access Denied Error.

To correct this error, return to the :guilabel:`OAuth consent screen` under

> :guilabel:`APIs & Services` and add test user(s) to the app. The email added should be the one

Reviewer 3: Line 198-199: replace the last sentence `The email added should be the one being configured to send in Odoo.` with `Add the email that you want to configure in Odoo.`

being configured to send in Odoo.

> Gmail Module not Updated

Reviewer 3: Line 201: `Updated` should be all lowercase

------------------------

> Should the :guilabel:`Gmail Module` in Odoo not be updated to the latest version, then a Forbidden

Reviewer 3: * replace `Should` with `If`
* add `has` after `in Odoo`
* add a `:guilabel:` to `Forbidden`

> Error message (You don't have the permission to access the requested resource. It is either

Reviewer 3: * make `Error` all lowercase
* (Line 205 - 206) delete the two sentences in parentheses

> read-protected or not readable by the server.) will populate.

Reviewer 3: * delete `will`
* replace `populate` with `populates`

.. image:: google_oauth/forbidden-error.png
   :class: img-fluid
   :align: center
   :alt: Forbidden Error in Google OAuth.

> To correct this error, go to the :guilabel:`Apps` module and clear the search terms out. Search for

Reviewer 3: * move the word `out` from the end of the sentence to after the word `clear`
* add the word `Then,` 
* de-capitalize `Search`

> *Gmail* or *Google* and upgrade the :guilabel:`Google Gmail` module. Click on the three dots on the

Reviewer 3: * add the word `Finally,` before `Click`
* de-capitalize the word `Click`

upper right of the module and select :guilabel:`Upgrade`.

> Desktop App VS Web Application Application Type

Reviewer 3: Line 217: delete `Desktop App VS Web Application` so just `Application Type` is left

-----------------------------------------------

> Should :guilabel:`Desktop App` under the :guilabel:`Application Type` be selected when creating the

Reviewer 3: Line 220-221: replace `Should :guilabel:'Desktop App' under the :guilabel:'Application Type' be selected when creating the :guilabel:'Credentials' (OAuth client ID)` with `When creating the :guilabel:'Credentials' (OAuth Client ID), if :guilabel:'Desktop App' is selected for the :guilabel:'Application Type'`

:guilabel:`Credentials` (OAuth client ID), the following error message will display:

.. image:: google_oauth/error-400.png
   :class: img-fluid
   :align: center
   :alt: Error 400 Redirect URI Mismatch.

To correct this error, delete the credentials already created and create new credentials using the

> 'Web Application' Application Type.

Reviewer 3: Line 228-229: replace `using the 'Web Application' Application Type` with `, selecting :guilabel:'Web Application' for the :guilabel:'Application Type'`

> .. image:: google_oauth/web-application.png

Reviewer 3: This image can be deleted.

   :class: img-fluid
   :align: center
   :alt: Change to Web Application in Google OAuth.

> Under :guilabel:`Authorized redirect URIs`, ADD URI:

Reviewer 3: * delete the line breaks before this so Line 229 and Line 236 are in one paragraph
* add the word `Then,` before `Under` and de-capitalize `Under`
